src/__tests__/posthog-core.js

```diff
@@ -677,24 +678,121 @@ describe('_loaded()', () => {
         })
     })
 
-    it('captures pageview', () => {
-        given('config', () => ({
-            capture_pageview: true,
-        }))
+    describe('capturing pageviews', () => {
+        beforeEach(() => {
+            given('config', () => ({
+                capture_pageview: false,
+                capture_performance: false, // the default
+            }))
+        })
 
-        given.subject()
+        it('captures not capture pageview if disabled', () => {
+            given.subject()
 
-        expect(given.overrides.capture).toHaveBeenCalledWith('$pageview', {}, { send_instantly: true })
-    })
+            expect(given.overrides.capture).not.toHaveBeenCalled()
+        })
 
-    it('captures not capture pageview if disabled', () => {
-        given('config', () => ({
-            capture_pageview: false,
-        }))
+        describe('when window performance is available', () => {
+            it('captures pageview', () => {
+                given('config', () => ({
+                    capture_pageview: true,
+                }))
 
-        given.subject()
+                given.subject()
 
-        expect(given.overrides.capture).not.toHaveBeenCalled()
+                expect(given.overrides.capture).toHaveBeenCalledWith('$pageview', {}, { send_instantly: true })
+                expect(window.performance.getEntriesByType).not.toHaveBeenCalled()
+            })
+
+            it('captures pageview with performance when enabled', () => {
+                given('config', () => ({
+                    capture_pageview: true,
+                    capture_performance: true,
+                }))
+
+                given.subject()
+
+                expect(given.overrides.capture).toHaveBeenCalledWith(
+                    '$pageview',
+                    { performance: { navigation: [], paint: [], resource: [] } },
+                    { send_instantly: true }
+                )
+                expect(window.performance.getEntriesByType).toHaveBeenCalledTimes(3)
+                expect(window.performance.getEntriesByType).toHaveBeenNthCalledWith(1, 'navigation')
+                expect(window.performance.getEntriesByType).toHaveBeenNthCalledWith(2, 'paint')
+                expect(window.performance.getEntriesByType).toHaveBeenNthCalledWith(3, 'resource')
+            })
+        })
+
+        describe('when window performance is not available', () => {
+            let performance
+            beforeEach(() => {
+                performance = { ...window.performance }
+
+                given('config', () => ({
```

Comment:
This does not need to be in a beforeEach. See also https://github.com/tatyshev/given2#more-examples